tasks/backfill_gh_app_installations.py

```diff
@@ -0,0 +1,149 @@
+import logging
+from typing import List, Optional
+
+import shared.torngit as torngit
+from sqlalchemy import String
+from sqlalchemy.orm.session import Session
+
+from app import celery_app
+from celery_config import backfill_gh_app_installations
+from database.models.core import GithubAppInstallation, Owner, Repository
+from services.owner import get_owner_provider_service
+from tasks.base import BaseCodecovTask
+
+log = logging.getLogger(__name__)
+
+
+class BackfillGHAppInstallationsTask(
+    BaseCodecovTask, name=backfill_gh_app_installations
+):
+    # Looping and adding all repositories in the installation app
+    async def add_repos_service_ids_from_provider(
+        self,
+        db_session: Session,
+        ownerid: int,
+        owner_service: torngit.base.TorngitBaseAdapter,
+        gh_app_installation: GithubAppInstallation,
+    ):
+        repos = await owner_service.list_repos_using_installation()
+        if repos:
+            # Fetching all repos service ids we have for that owner in the DB
+            repo_service_ids_in_db = [
+                repo.service_id
+                for repo in db_session.query(Repository.service_id)
+                .filter_by(ownerid=ownerid)
+                .all()
+            ]
+
+            # Add service ids from provider that we have DB records for to a list
+            new_repo_service_ids = set()
+            current_repo_service_ids = set(
+                gh_app_installation.repository_service_ids or []
+            )
+            for repo in repos:
+                repo_data = repo["repo"]
+                service_id = repo_data["service_id"]
+                if service_id and service_id in repo_service_ids_in_db:
+                    new_repo_service_ids.add(service_id)
+            log.info(
+                "Added the following repo service ids to this gh app installation",
+                extra=dict(
+                    ownerid=ownerid,
+                    installation_id=gh_app_installation.installation_id,
+                    new_repo_service_ids=list(new_repo_service_ids),
+                ),
+            )
+            gh_app_installation.repository_service_ids = list(
```

Comment:
I think by definition we only want the new ones. If GH doesn't tell us we can access the repo and Codecov doesn't have the repo in the db, then we don't want it in the list. I think the union with current is redundant. Thoughts?

Reply:
So the question is, can anything be in current that's not in new. I think most cases would be redundant, like `{1}.union({1,2,3}) = {1,2,3}`. We could get something like `{1}.union({2,3}) = {1,2,3}`, but by definition that would mean a repo that was part of an org (or selected by the app) that is no longer, so yeah I think you'd only want the new, not new + old. I'll make a change

Reply:
This sounds good.